packets.go

```diff
@@ -1078,17 +1078,19 @@ func (stmt *mysqlStmt) writeExecutePacket(args []driver.Value) error {
 				paramTypes[i+i] = fieldTypeString
 				paramTypes[i+i+1] = 0x00
 
-				var val []byte
+				var a [64]byte
+				var b = a[:0]
```

Comment:
Sorry for commenting on old piece of code but I came here from an article and I can't help but wonder, why this wasn't done by `make([]byte, 0, 64)`. Isn't it the same?

Reply:
I think the compiler can optimize “make” to be stack allocated if possible, so the end result is the same whether we use make or declare an array, it’s just a style decision. 

The optimization doesn’t come from the use of an array, it comes from writing directly into a pre-allocated buffer instead of working with intermediary heap allocated values.